### PR TITLE
A few impovements to benchmarks

### PR DESCRIPTION
**Pull request by rth**
Following discussion in https://github.com/iodide-project/pyodide/pull/202 this,
 - improves PEP8 for the benchmark files
 - uses the vectorized numpy expressions in some benchmarks
 - tries to decrease the overall run time from 40min to ~10min so it doesn't take longer than tests.
 - also fixes a few interaction between benchmarks & conftest.py that got broken by the last couple of quasi simultaneous PRs.

more comments below,

### CODE REVIEW

File: benchmark/benchmarks/conv.py [hunk 1]
```diff
@@ -1,27 +0,0 @@
...
-def conv(x, weights):
-    sx = x.shape
-    sw = weights.shape
-    result = np.zeros_like(x)
-    for i in range(sx[0]):
-        for j in range(sx[1]):
-            for ii in range(sw[0]):
-                for jj in range(sw[1]):
-                    idx = clamp(i,ii-sw[0]//2,sw[0]), clamp(j,jj-sw[0]//2,sw[0])
-                    result[i,j] += x[idx] * weights[ii,jj]
```
Comment (rth) [inline benchmark/benchmarks/conv.py]: Removed this as we currently have a number of such N level loops over numpy arrays (e.g. `pairwise_loop.py`) and I don't think this provides much additional information. Saying that it's a convolution benchmark is confusing, as it certainly unrelated to `np.convolve`. Maybe we should add the latter as a benchmark.

File: benchmark/benchmarks/wdist.py [hunk 2]
```diff
@@ -1,18 +0,0 @@
...
-def wdist(A, B, W):
-
-    k,m = A.shape
-    _,n = B.shape
-    D = np.zeros((m, n))
-
-    for ii in range(m):
-        for jj in range(n):
-            wdiff = (A[:,ii] - B[:,jj]) / W[:,ii]
-            D[ii,jj] = np.sqrt((wdiff**2).sum())
```
Comment (rth) [inline benchmark/benchmarks/wdist.py]: This performs very similar computations to `allpairs_distances_loops.py` so I removed it.

**Comment by mdboom** [comment]
I'm not seeing significant changes in the results with this changes, even with the smaller data sizes, which is great.

**Comment by rth** [comment]
> Now that the benchmarks are PEP8-compliant, let's fix make lint to check them.

They were not, it was just a first pass. Now re-run autopep8 on them and enabled linting -- should be good now.

> I'm not seeing significant changes in the results with this changes, even with the smaller data sizes, which is great.

I was hoping they wouldn't.


> (And probably do the numpy one on a larger data size, since we only care about native vs. wasm speed anyway).

Done.

> This actually might be a cool one to implement both ways: using Numpy and not. Maybe split into "python_fft" and "numpy_fft"? 

So on one level I agree that this trick with recursive fft calculation is cool. However I think that it's important for benchmarks to represent real use cases so that people who look at them can imagine what consequences using pyodide would have for them. In the case of pure python fft (or doing nested loops element by element over numpy arrays, in general), I cannot think of a real-world use-case, except to demonstrate that such code can be efficiently optimized with Cython/Pythran/numba. On the other hand, I think that it would be great to have,
 - more pure python benchmarks that represent real-use cases. For instance PyPy has a selection in http://speed.pypy.org/changes/  Maybe we could adapt something
 - some pandas and matplotlib bechmarks  (though it would require some workarounds for Chrome for the former).

What do you think?

**Pull request merged**